doc/src/manual/parallel-computing.md

```diff
@@ -1,12 +1,546 @@
-2. Shared memory using multiple processes on a single node
-3. Multi-threading
+1. Julia Coroutines (Green Threading)
+2. Multi-Threading
+3. Multi-Core or Distributed Processing
 
-# Distributed Memory Parallelism
+We will consider first Julia [Tasks (aka Coroutines)](@ref man-tasks) and other high-level interfaces and 
+modules that rely on the Julia runtime library, that allow to suspend and resume computations with full 
+control of inter-`Tasks` communication without worrying of the opeartive system's scheduler. Julia also
+allows to communicate between `Tasks` through operations like [`wait`](@ref) and [`fetch`](@ref).
+
+Julia also supports multi-threading, where execution is forked and an anonymous function is run across all
+threads. Described as a fork-join approach, all threads have to join together and serial execution continues.
+Multi-threading is supported using the `Base.Threads` module that is still considered experimental, as Julia is
+not fully thread-safe, consider for example [this issue](https://github.com/JuliaLang/julia/issues/22581).
+Also multi-threading should be implemented only if you take into consideration global variables, locks and 
+atomics, so we will present it later.
+
+In the end we will present Julia's way to distributed and parallel computing. With scientific computing 
+in mind, Julia natively implements interfaces to send a process through multiple cores or machines.
+Also we will mention useful external packages for distributed programming like `MPI.jl` and `DistributeArrays.jl`.
+
+
+# Coroutines
+
+Julia's parallel programming platform uses [Tasks (aka Coroutines)](@ref man-tasks) to switch among multiple
+computations. Whenever code performs a communication operation like [`fetch`](@ref) or [`wait`](@ref),
+the current task is suspended and a scheduler picks another task to run. A task is restarted when
+the event it is waiting for completes.
+
+For many problems, it is not necessary to think about tasks directly. However, they can be used
+to wait for multiple events at the same time, which provides for *dynamic scheduling*. In dynamic
+scheduling, a program decides what to compute or where to compute it based on when other jobs
+finish. This is needed for unpredictable or unbalanced workloads, where we want to assign more
+work to processes only when they finish their current tasks.
+
+As an example, consider computing the singular values of matrices of different sizes:
+
+```julia-repl
+julia> M = Matrix{Float64}[rand(800,800), rand(600,600), rand(800,800), rand(600,600)];
+
+julia> pmap(svd, M);
+```
+
+If one process handles both 800×800 matrices and another handles both 600×600 matrices, we will
+not get as much scalability as we could. The solution is to make a local task to "feed" work to
+each process when it completes its current task. For example, consider a simple [`pmap`](@ref)
+implementation:
+
+```julia
+function pmap(f, lst)
```

Comment:
Since this is now changed the order (for the better IMO), I think changing this example would be worthwhile.
This example is uses concepts from the distributed section and thus might be confusing to users, also in my
mind it is primarily about how to use `remotecall_fetch` or how to run blocking distributed operations.